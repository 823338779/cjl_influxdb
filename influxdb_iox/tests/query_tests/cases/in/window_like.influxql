-- Query tests derived from https://github.com/influxdata/influxdb_iox/issues/6112
-- IOX_SETUP: window_like

--
-- difference
--
SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s);
SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(none);
SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(0);
SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(previous);
SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(linear);

--
-- moving_average
--
SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s);
SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(none);
SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(0);
SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(previous);
SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(linear);