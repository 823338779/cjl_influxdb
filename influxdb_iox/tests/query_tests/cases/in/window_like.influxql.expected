-- Test Setup: window_like
-- InfluxQL: SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s);
name: diskio
+---------------------+------------+-----------+
| time                | difference | mean      |
+---------------------+------------+-----------+
| 2020-06-11T16:52:54 |            | 5592646.0 |
| 2020-06-11T16:53:01 |            |           |
| 2020-06-11T16:53:08 | 164.0      | 5592810.0 |
| 2020-06-11T16:53:15 | 187.0      | 5592997.0 |
| 2020-06-11T16:53:22 |            |           |
| 2020-06-11T16:53:29 | 112.0      | 5593109.0 |
| 2020-06-11T16:53:36 | 110.0      | 5593219.0 |
| 2020-06-11T16:53:43 |            |           |
| 2020-06-11T16:53:50 | 219.0      | 5593438.0 |
| 2020-06-11T16:53:57 | 75.0       | 5593513.0 |
| 2020-06-11T16:54:04 | 76.0       | 5593589.0 |
| 2020-06-11T16:54:11 |            |           |
| 2020-06-11T16:54:18 | 146.0      | 5593735.0 |
| 2020-06-11T16:54:25 |            |           |
| 2020-06-11T16:54:32 |            |           |
| 2020-06-11T16:54:39 |            |           |
| 2020-06-11T16:54:46 |            |           |
| 2020-06-11T16:54:53 |            |           |
+---------------------+------------+-----------+
-- InfluxQL: SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(none);
name: diskio
+---------------------+------------+-----------+
| time                | difference | mean      |
+---------------------+------------+-----------+
| 2020-06-11T16:52:54 |            | 5592646.0 |
| 2020-06-11T16:53:08 | 164.0      | 5592810.0 |
| 2020-06-11T16:53:15 | 187.0      | 5592997.0 |
| 2020-06-11T16:53:29 | 112.0      | 5593109.0 |
| 2020-06-11T16:53:36 | 110.0      | 5593219.0 |
| 2020-06-11T16:53:50 | 219.0      | 5593438.0 |
| 2020-06-11T16:53:57 | 75.0       | 5593513.0 |
| 2020-06-11T16:54:04 | 76.0       | 5593589.0 |
| 2020-06-11T16:54:18 | 146.0      | 5593735.0 |
+---------------------+------------+-----------+
-- InfluxQL: SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(previous);
name: diskio
+---------------------+------------+-----------+
| time                | difference | mean      |
+---------------------+------------+-----------+
| 2020-06-11T16:52:54 |            | 5592646.0 |
| 2020-06-11T16:53:01 | 0.0        | 5592646.0 |
| 2020-06-11T16:53:08 | 164.0      | 5592810.0 |
| 2020-06-11T16:53:15 | 187.0      | 5592997.0 |
| 2020-06-11T16:53:22 | 0.0        | 5592997.0 |
| 2020-06-11T16:53:29 | 112.0      | 5593109.0 |
| 2020-06-11T16:53:36 | 110.0      | 5593219.0 |
| 2020-06-11T16:53:43 | 0.0        | 5593219.0 |
| 2020-06-11T16:53:50 | 219.0      | 5593438.0 |
| 2020-06-11T16:53:57 | 75.0       | 5593513.0 |
| 2020-06-11T16:54:04 | 76.0       | 5593589.0 |
| 2020-06-11T16:54:11 | 0.0        | 5593589.0 |
| 2020-06-11T16:54:18 | 146.0      | 5593735.0 |
| 2020-06-11T16:54:25 | 0.0        | 5593735.0 |
| 2020-06-11T16:54:32 | 0.0        | 5593735.0 |
| 2020-06-11T16:54:39 | 0.0        | 5593735.0 |
| 2020-06-11T16:54:46 | 0.0        | 5593735.0 |
| 2020-06-11T16:54:53 | 0.0        | 5593735.0 |
+---------------------+------------+-----------+
-- InfluxQL: SELECT difference(mean(writes)), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(linear);
name: diskio
+---------------------+------------+-----------+
| time                | difference | mean      |
+---------------------+------------+-----------+
| 2020-06-11T16:52:54 |            | 5592646.0 |
| 2020-06-11T16:53:01 | 82.0       | 5592728.0 |
| 2020-06-11T16:53:08 | 82.0       | 5592810.0 |
| 2020-06-11T16:53:15 | 187.0      | 5592997.0 |
| 2020-06-11T16:53:22 | 56.0       | 5593053.0 |
| 2020-06-11T16:53:29 | 56.0       | 5593109.0 |
| 2020-06-11T16:53:36 | 110.0      | 5593219.0 |
| 2020-06-11T16:53:43 | 109.5      | 5593328.5 |
| 2020-06-11T16:53:50 | 109.5      | 5593438.0 |
| 2020-06-11T16:53:57 | 75.0       | 5593513.0 |
| 2020-06-11T16:54:04 | 76.0       | 5593589.0 |
| 2020-06-11T16:54:11 | 73.0       | 5593662.0 |
| 2020-06-11T16:54:18 | 73.0       | 5593735.0 |
| 2020-06-11T16:54:25 |            |           |
| 2020-06-11T16:54:32 |            |           |
| 2020-06-11T16:54:39 |            |           |
| 2020-06-11T16:54:46 |            |           |
| 2020-06-11T16:54:53 |            |           |
+---------------------+------------+-----------+
-- InfluxQL: SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(none);
name: diskio
+---------------------+-------------------+-----------+
| time                | moving_average    | mean      |
+---------------------+-------------------+-----------+
| 2020-06-11T16:52:54 |                   | 5592646.0 |
| 2020-06-11T16:53:08 |                   | 5592810.0 |
| 2020-06-11T16:53:15 | 5592817.666666667 | 5592997.0 |
| 2020-06-11T16:53:29 | 5592972.0         | 5593109.0 |
| 2020-06-11T16:53:36 | 5593108.333333333 | 5593219.0 |
| 2020-06-11T16:53:50 | 5593255.333333333 | 5593438.0 |
| 2020-06-11T16:53:57 | 5593390.0         | 5593513.0 |
| 2020-06-11T16:54:04 | 5593513.333333333 | 5593589.0 |
| 2020-06-11T16:54:18 | 5593612.333333333 | 5593735.0 |
+---------------------+-------------------+-----------+
-- InfluxQL: SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(previous);
name: diskio
+---------------------+-------------------+-----------+
| time                | moving_average    | mean      |
+---------------------+-------------------+-----------+
| 2020-06-11T16:52:54 |                   | 5592646.0 |
| 2020-06-11T16:53:01 |                   | 5592646.0 |
| 2020-06-11T16:53:08 | 5592700.666666667 | 5592810.0 |
| 2020-06-11T16:53:15 | 5592817.666666667 | 5592997.0 |
| 2020-06-11T16:53:22 | 5592934.666666667 | 5592997.0 |
| 2020-06-11T16:53:29 | 5593034.333333333 | 5593109.0 |
| 2020-06-11T16:53:36 | 5593108.333333333 | 5593219.0 |
| 2020-06-11T16:53:43 | 5593182.333333333 | 5593219.0 |
| 2020-06-11T16:53:50 | 5593292.0         | 5593438.0 |
| 2020-06-11T16:53:57 | 5593390.0         | 5593513.0 |
| 2020-06-11T16:54:04 | 5593513.333333333 | 5593589.0 |
| 2020-06-11T16:54:11 | 5593563.666666667 | 5593589.0 |
| 2020-06-11T16:54:18 | 5593637.666666667 | 5593735.0 |
| 2020-06-11T16:54:25 | 5593686.333333333 | 5593735.0 |
| 2020-06-11T16:54:32 | 5593735.0         | 5593735.0 |
| 2020-06-11T16:54:39 | 5593735.0         | 5593735.0 |
| 2020-06-11T16:54:46 | 5593735.0         | 5593735.0 |
| 2020-06-11T16:54:53 | 5593735.0         | 5593735.0 |
+---------------------+-------------------+-----------+
-- InfluxQL: SELECT moving_average(mean(writes), 3), mean(writes) FROM diskio WHERE time >= '2020-06-11T16:53:00Z' AND time < '2020-06-11T16:55:00Z' GROUP BY time(7s) fill(linear);
name: diskio
+---------------------+-------------------+-----------+
| time                | moving_average    | mean      |
+---------------------+-------------------+-----------+
| 2020-06-11T16:52:54 |                   | 5592646.0 |
| 2020-06-11T16:53:01 |                   | 5592728.0 |
| 2020-06-11T16:53:08 | 5592728.0         | 5592810.0 |
| 2020-06-11T16:53:15 | 5592845.0         | 5592997.0 |
| 2020-06-11T16:53:22 | 5592953.333333333 | 5593053.0 |
| 2020-06-11T16:53:29 | 5593053.0         | 5593109.0 |
| 2020-06-11T16:53:36 | 5593127.0         | 5593219.0 |
| 2020-06-11T16:53:43 | 5593218.833333333 | 5593328.5 |
| 2020-06-11T16:53:50 | 5593328.5         | 5593438.0 |
| 2020-06-11T16:53:57 | 5593426.5         | 5593513.0 |
| 2020-06-11T16:54:04 | 5593513.333333333 | 5593589.0 |
| 2020-06-11T16:54:11 | 5593588.0         | 5593662.0 |
| 2020-06-11T16:54:18 | 5593662.0         | 5593735.0 |
| 2020-06-11T16:54:25 |                   |           |
| 2020-06-11T16:54:32 |                   |           |
| 2020-06-11T16:54:39 |                   |           |
| 2020-06-11T16:54:46 |                   |           |
| 2020-06-11T16:54:53 |                   |           |
+---------------------+-------------------+-----------+